.github/workflows/ubuntu.yml

```diff
@@ -166,33 +162,34 @@ jobs:
       - name: configure
         run: |
           cd _build
-          ../configure --enable-code-coverage CPPFLAGS="-Werror=declaration-after-statement"
+          ../configure --enable-code-coverage \
+                       CPPFLAGS="-Werror=declaration-after-statement -pedantic" \
+                       CC="gcc -std=c89"
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         if: failure()
         with:
-          name: config.log
+          name: config-${{ matrix.os }}.log
           path: _build/config.log
 
       - name: make
         run: |
-          cd _build
-          make --jobs=$(($(nproc)+1))
+          make -C _build --jobs=$(($(nproc)+1))
 
       - name: coverage
         run: |
-          cd _build
-          make check-code-coverage TESTSUITEFLAGS="--jobs=$(($(nproc)+1))"
+          make -C _build check-code-coverage \
+                         TESTSUITEFLAGS="--jobs=$(($(nproc)+1))"
```

Comment:
Can we have both the "normal" and the extended coverage?
Also: should we use a two step approach for generated the "normal one"?

As an explanation for both - here's what we use in Appveyor:

```sh
# note: 'check-code-coverage' ignores errors, so doing it as two-step manually
# make check-code-coverage TESTSUITEFLAGS="--jobs=$(($(nproc)+1))" CODE_COVERAGE_DIRECTORY=$(realpath .)
make check TESTSUITEFLAGS="--jobs=$(($(nproc)+1))"
make code-coverage-capture CODE_COVERAGE_DIRECTORY=$(realpath .)
appveyor AddMessage "internal testsuite and coverage check are finished" -Details "$(grep -A3 "Lines:\|Functions:\|Branches:" GnuCOBOL-$VERSION-coverage/index.html | sed -e "s/.*>\(.*\)<.*/\1/g" | tr '\n' ' ')"
zip -r GnuCOBOL-$VERSION-coverage.zip GnuCOBOL-$VERSION-coverage
appveyor PushArtifact GnuCOBOL-$VERSION-coverage.zip
make -k --jobs=$(($(nproc)+1)) -C tests/cobol85 test
# TODO: add NIST test summary in details
appveyor AddMessage "NIST testsuite is finished"
make code-coverage-capture CODE_COVERAGE_OUTPUT_DIRECTORY=extended-coverage CODE_COVERAGE_OUTPUT_FILE=extended-coverage.info CODE_COVERAGE_DIRECTORY=$(realpath .)
appveyor AddMessage "extended coverage check is finished" -Details "$(grep -A3 "Lines:\|Functions:\|Branches:" extended-coverage/index.html | sed -e "s/.*>\(.*\)<.*/\1/g" | tr '\n' ' ')"
zip -r GnuCOBOL-$VERSION-coverage-extended.zip extended-coverage
appveyor PushArtifact GnuCOBOL-$VERSION-coverage-extended.zip
```